en/components/grid.md

```diff
@@ -75,39 +73,127 @@ Let's turn the **autoGenerate** property off and define the columns collection i
+
+Each of the column templates can be changed programmatically at any point through the column object itself. For example in the code below, we have declared two templates for our user data. In our Typescript code we'll get references to the templates themselves and then based on some condition we will render the appropriate template for the column in our application.
+
+```html
+<igx-grid>
+    <!-- Column declarations -->
+</igx-grid>
+
+<ng-template #normalView let-value>
+    <div class="user-details">{{ val }}</div>
+    <user-details-component></user-details-component>
+</ng-template>
+
+<ng-template #smallView let-value>
+    <div class="user-details-small">{{ val }}</div>
+</ng-template>
+```
 
 ```typescript
-public initColumns(event: IgxGridColumnInitEvent) {
-    const column: IgxColumnComponent = event.column;
+@ViewChild("normalView", { read: TemplateRef })
+public normalView: TemplateRef<any>;
+
+@ViewChild("smallView", { read: TemplateRef })
+public smallView: TemplateRef<any>;
+
+....
+
+const column = this.grid.getColumnByName("User");
+// Return the appropriate template based on some conditiion.
+// For example saved user settings, viewport size, etc.
+column.bodyTemplate = this.smallView;
+```
+
+Column properties can also be set in code in the **initColumns** event which is emitted when the columns are initialized in the grid.
+
+```typescript
+public initColumns(column: IgxGridColumn) {
+    const column: IgxColumnComponent = column;
     if (column.field === 'ProductName') {
-      column.filtering = true;
-      column.sortable = true;
-      column.editable = true;
+        column.filterable = true;
+        column.sortable = true;
+        column.editable = true;
     }
 }
 ```
 
-The code above will make the **ProductName** column sortable, filterable, and editable and will instantiate the corresponding features UI (like inputs for editing, save dialogs, etc.).
-
-<div class="divider--half"></div>
+The code above will make the **ProductName** column sortable, filterable, and editable and will instantiate the corresponding features UI (like inputs for editing, etc.).
 
 ### Data binding
 
-Before going any further with the grid we want to change the grid to bind to remote data service, which is a common scenario in real production applications. A good practice is to separate all data fetching related logic in a separate data service, so we are going to create a service which
-will handle the fetching of data from the server.
+Before going any further with the grid we want to change the grid to bind to remote data service, which is a common scenario in real production applications. A good practice is to separate all data fetching related logic in a separate data service, so we are going to create a service which will handle the fetching of data from the server.
```

Comment:
Don't use language such as "real production applications" - this is internal language. Use something like - "large-scale applications", "enterprise-grade applications".